PlayerKit/Classes/RegularPlayer.swift

```diff
@@ -168,8 +168,51 @@ extension AVMediaSelectionOption: TextTrackMetadata {
     private func setupAirplay() {
         self.player.usesExternalPlaybackWhileExternalScreenIsActive = true
     }
+
+    // MARK: - Smooth Seeking
+
+    // Note: Smooth seeking follows the guide from Apple Technical Q&A: https://developer.apple.com/library/archive/qa/qa1820/_index.html
+    // Update the seek target and begin seeking if there is no seek currently in progress.
+    private func smoothSeek(to cmTime: CMTime) {
+        self.seekTarget = cmTime
+
+        guard self.isSeekInProgress == false else { return }
+        self.seekToTarget()
+    }
+
+    // Unconditionally seek to the current seek target.
+    private func seekToTarget() {
+        assert(!self.isSeekInProgress)
+        self.isSeekInProgress = true
+
+        guard self.player.status != .unknown else { return }
+
+        assert(CMTIME_IS_VALID(self.seekTarget))
+        let inProgressSeekTarget = self.seekTarget
+
+        let completion: (Bool) -> Void = { _ in
+            self.time = CMTimeGetSeconds(inProgressSeekTarget)
```

Comment:
If we want to be extra safe, we can weakly capture `self` here.

Reply:
I think this is safe--`completion` will be called whether or not the seek completes successfully, so `AVPlayer` will not indefinitely retain it. I guess it can't hurt to be sure...